### PR TITLE
"CTRP/CTRO Selenium Automation" Chaudhryma

### PR DESCRIPTION
**Pull request by Mariachaudhry**

### CODE REVIEW

File: src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Features/CTRP_CTRO_NV.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Feature: CTRP/CTRO Native View Case Test Scenarios
+
+  @CTRPSNOW-81 @chaudhryma
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Features/CTRP_CTRO_NV.feature:3]: Please add @Regression, @Smoke tags as needed. Or add @InProgress if you are still working on it

File: src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Selenium_Runners/Run_CTRP_Progression_Test.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package ServiceNow.PlatformBusinessApps.CTRP_CTRO.Selenium.Selenium_Runners;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
+        "junit:target/cucumber.xml", "rerun:target/failed.txt",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Selenium_Runners/Run_CTRP_Progression_Test.java:7]: Please remove "junit:target/cucumber.xml"

File: src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Selenium_Runners/Run_CTRP_Progression_Test.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package ServiceNow.PlatformBusinessApps.CTRP_CTRO.Selenium.Selenium_Runners;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Selenium_Runners/Run_CTRP_Progression_Test.java:6]: Please add **.html** at the end like "html:target/html-reports/cucumber-default-report.html"

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java [hunk 4]
```diff
@@ -1,4 +1,64 @@
...
 
-public class Example {
+import ServiceNow.PlatformBusinessApps.NERD.selenium.Constants.TopAccomplishmentsSubmission_Constants;
+import appsCommon.PageInitializers.PageInitializer;
+import com.microsoft.playwright.*;
+import com.microsoft.playwright.options.*;
+import com.nci.automation.utils.MiscUtils;
+import com.nci.automation.web.CommonUtils;
+import com.nci.automation.web.WebDriverUtils;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java:10]: Please remove the empty line

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java [hunk 5]
```diff
@@ -1,4 +1,64 @@
...
+            page.navigate("https://service-test.nci.nih.gov/side_door.do");
+            page.getByLabel("User name").click();
+            page.getByLabel("User name").fill("CBIITTestAccount");
+            page.getByLabel("Password", new Page.GetByLabelOptions().setExact(true)).click();
+            page.getByLabel("Password", new Page.GetByLabelOptions().setExact(true)).fill("curlbylawdighalvekinlarswould");
+            page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Log in")).click();
+            page.navigate("https://service-test.nci.nih.gov/nerd");
+//            page.locator("#single-button").first().click();
+//            page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Signed in as CBIIT Test")).click();
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java:32]: Please remove empty line 32

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java [hunk 6]
```diff
@@ -1,4 +1,64 @@
...
+            // Switch to the form's context (optional)
+            // Locate the form element
+            CommonUtils.switchToFrame(createNewSubmissionPage.DescriptionTextBoxIframe);
+//            page1.frameLocator("#ui-tinymce-5461_ifr").locator("html").click();
+//            page1.frameLocator("#ui-tinymce-5461_ifr").getByLabel("Rich Text Area. Press ALT-0").fill("Test");
+            createNewSubmissionPage.DescriptionTextBox.sendKeys(TopAccomplishmentsSubmission_Constants.TOP_ACCOMPLISHMENTS_DESCRIPTION_TEXT_BOX);
+            MiscUtils.sleep(2000);
+            WebDriverUtils.webDriver.switchTo().defaultContent();
+            MiscUtils.sleep(2000);
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java:51]: Please remove empty line 51

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java [hunk 7]
```diff
@@ -1,4 +1,64 @@
...
+//            const formElement = await page1.$('form');
+//            await page1.focus('form');
+//            page1.frameLocator("#ui-tinymce-528_ifr").getByRole(AriaRole.PARAGRAPH).click();
+//            page1.frameLocator("#ui-tinymce-8544_ifr").getByLabel("Rich Text Area. Press ALT-0").fill("Description");
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java:55]: Please uncomment lines 52-55 or remove them if no longer needed

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java [hunk 8]
```diff
@@ -1,4 +1,64 @@
...
+            page1.locator("select[name=\"x_26385_crs_kd_u_kb_research_type\"]").selectOption("Cancer Control");
+            page1.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit")).click();
+            page1.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Yes")).click();
+            page1.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Ok")).click();
+
+        }
+    }
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/playwright/Steps/Example.java:63]: Please remove empty line 63

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature [hunk 9]
```diff
@@ -15,4 +15,11 @@ Feature: NCI Staff Member Test Scenarios
       | knowledgeBaseOne | knowledgeBaseTwo | knowledgeBaseThree  | knowledgeCategory     |
       | NERD             | ROCK             | Moonshot Evaluation | Top Accomplishments   |
       | NERD             | ROCK             | Moonshot Evaluation | Other Accomplishments |
-      | NERD             | ROCK             | Moonshot Evaluation | Collaborations        |
\ No newline at end of file
+      | NERD             | ROCK             | Moonshot Evaluation | Collaborations        |
+
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature:20]: Please remove the empty line

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature [hunk 10]
```diff
@@ -15,4 +15,11 @@ Feature: NCI Staff Member Test Scenarios
...
+  @NERDSNOW-686 @chaudhryma @Progression @In_Progress
+  Scenario: Test 1: Remove 'Delete' Button in NERD For Specific Users
+    Given an NCI Staff Member not on the NERD table is on the NERD CRS Knowledge Base
+    When they view Top Accomplishments submission category
+#    Then they DO NOT see the 'Delete' button for non-Published articles
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature:25]: Please remove the commented line 25

File: src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java [hunk 11]
```diff
@@ -6,8 +6,8 @@
 @CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
         "junit:target/cucumber.xml", "rerun:target/failed.txt",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java:7]: Please remove "junit:target/cucumber.xml"

File: src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java [hunk 12]
```diff
@@ -6,8 +6,8 @@
 @CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java:6]: Please add **.html** at the end like "html:target/html-reports/cucumber-default-report.html"

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Please review the comments. Thanks!

File: src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Features/CTRP_CTRO_NV.feature (hunk 1 above)
Comment (matakevin) [reply src/test/java/ServiceNow/PlatformBusinessApps/CTRP_CTRO/Selenium/Features/CTRP_CTRO_NV.feature:3]: I agree with this comment

**Review by matakevin — CHANGES_REQUESTED** [review]
Please make the changes Hamid requested.  He seems to have beaten me to all the suggested changes, thank you!

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature (hunk 10 above)
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Features/NCI_Staff_Members.feature:25]: In progress :)

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java [hunk 13]
```diff
@@ -148,4 +150,40 @@ public static void aProgramStaffMemberIsOnTheCRSKnowledgeManagementSystemPage(St
...
+        CucumberLogUtils.logScreenshot();
+        nerdCRSTopAccomplishmentsPage.nerdTopAccomplishmentsAccordionButton.click();
+        MiscUtils.sleep(1000);
+        CucumberLogUtils.logScreenshot();
+        JavascriptUtils.drawBlueBorder(nerdDynamicXpaths.underReviewText(TopAccomplishmentsSubmission_Constants.TOP_ACCOMPLISHMENTS_SUBMISSION_NAME));
+        MiscUtils.sleep(1000);
+        CucumberLogUtils.logScreenshot();
+        MiscUtils.sleep(20000);
+
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java]: Please remove empty lines186-187

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java [hunk 14]
```diff
@@ -148,4 +150,40 @@ public static void aProgramStaffMemberIsOnTheCRSKnowledgeManagementSystemPage(St
...
+        WebDriverUtils.webDriver.get(EnvUtils.getApplicationUrl("NERD"));
+        MiscUtils.sleep(1000);
+        JavascriptUtils.drawBlueBorder(
+                nerdKnowledgeBasePage.nerdCRSKnowledgeMainText);
+        JavascriptUtils.drawBlueBorder(
+                nerdCrsKnowledgeDatabaseSubmissionsPage.crsKnowledgeManagementSystemSubmissionsPageSubmissionsLink);
+        CucumberLogUtils.logScreenshot();
+        MiscUtils.sleep(2000);
+        NERDApplicationStepsImplementation.creatingNewSubmission(nerdCrsKnowledgeDatabaseSubmissionsPage.topAccomplishmentsCreateNewSubmissionLink);
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java:171]: Please remove the empty line

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+public class NERD_NCI_StaffMembers_Steps {
+    @Given("an NCI Staff Member not on the NERD table is on the NERD CRS Knowledge Base")
+    public void an_nci_staff_member_not_on_the_nerd_table_is_on_the_nerd_crs_knowledge_base() {
+        NERD_NCI_StaffMemberStepsImplementation.an_nci_staff_member_not_on_the_nerd_table_is_on_the_nerd_crs_knowledge_base();
+    }
+
+    @When("they view Top Accomplishments submission category")
+    public void they_view_top_accomplishments_submission_category() {
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java]: This step has not been implemented yet

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Pages/NERDCRSTopAccomplishmentsPage.java
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Pages/NERDCRSTopAccomplishmentsPage.java:1]: remove line 84

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java:1]: remove line 15 unless still in progress

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/StepsImplementation/NERD_NCI_StaffMemberStepsImplementation.java:1]: delete lines 165 and 186

File: src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java
Comment (matakevin) [inline src/test/java/ServiceNow/ServiceNowRunners/NERDRunners/Run_NERD_Progression_Test.java:1]: delete 20 and 21

**Review by matakevin — CHANGES_REQUESTED** [review]
Please check the requested changes

**Review by hamidnesar — CHANGES_REQUESTED** [review]

File: src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java (hunk 15 above)
Comment (Mariachaudhry) [reply src/test/java/ServiceNow/PlatformBusinessApps/NERD/selenium/Steps/NERD_NCI_StaffMembers_Steps.java]: in progress :)

**Review by matakevin — APPROVED** [review]
LGTM, thank you Maria!

**Review by georgeb79 — APPROVED** [review]

**Review by hamidnesar — APPROVED** [review]

**Pull request merged**